app/assets/javascripts/discourse/controllers/composer.js.es6

```diff
@@ -56,6 +62,19 @@ export default Ember.Controller.extend({
   topic: null,
   linkLookup: null,
 
+  init() {
+    this._super();
+
+    addPopupMenuOptionsCallback(function(controlller) {
+      return {
+        action: 'toggleWhisper',
+        icon: 'eye-slash',
+        label: 'composer.toggle_whisper',
+        condition: controlller.get("canWhisper")
```

Comment:
I am a little worried to be honest about passing the entire `controller` through to the API.

I understand why you did this, but it means in the future to support this API we have to support passing the entire controller through, which means also letting people call actions on it or whatever crazy thing plugin authors decide to do.

I'd prefer if it was called with an object that contains the attributes the plugin author would want to use. 

For example `{ canWhisper: true}`